lib/pdn/pdn.c

```diff
@@ -659,6 +659,51 @@ int pdn_dynamic_params_get(uint8_t cid, struct in_addr *dns4_pri,
 	return 0;
 }
 
+int pdn_dynamic_params_get_v6(uint8_t cid, struct in_addr *dns6_pri,
```

Comment:
Would it be possible to add the IPv6 functionality to pdn_dynamic_params_get, possibly with preference/family parameter?

There is very little difference between the two functions and it would be more efficient and better encapsulated, if there was a loop for checking IPv4 and IPv6 in one function.


Ping: @nrfconnect/ncs-modem

Reply:
I was thinking of that as well, by using `struct sockaddr` instead of `struct inaddr`, which contain both the family and the address.

Reply:
If we are changing the signature of the function there is also the option to add support for retrieving multiple `+CGCONTRDP` entries, to get both IPv4 and IPv6 dns addresses.

Reply:
I also thought this but I hesitated, because I didn't want to break something on the existing one.

Reply:
Adding a new function for this is an improvement, so this works for me even without combination of functionality.